### ISSUE TITLE
Regression: git dependency with incorrect package name causes infinite loop in resolver

### ISSUE THREAD

**Issue by frankier** [sdispater/poetry#1064]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **Debian buster**: 
- **0.12.14**: 

## Issue

I think this is a regression to https://github.com/sdispater/poetry/issues/259

Steps to reproduce:

 $ pyenv shell 3.7.2
 $ poetry init # choose defaults but don't add anything
 $ poetry add -vv --git https://github.com/frankier/finn-sense-clust.git finn-sense-clust # for example -- module name should be senseclust

Output:

...
   1: selecting senseclust (0.0.1 bbd609e)
   1: fact: senseclust (0.0.1) depends on nltk (*)
   1: fact: senseclust (0.0.1) depends on sklearn (*)
   1: fact: senseclust (0.0.1) depends on numpy (*)
   1: fact: senseclust (0.0.1) depends on scipy (*)
   1: fact: senseclust (0.0.1) depends on matplotlib (*)
   1: fact: senseclust (0.0.1) depends on click (>=7)
   1: fact: senseclust (0.0.1) depends on finntk (>=0.0.14)
   1: fact: senseclust (0.0.1) depends on fastcluster (*)
   1: fact: senseclust (0.0.1) depends on networkx (*)
   1: fact: senseclust (0.0.1) depends on wikiparse (*)
   1: fact: senseclust (0.0.1) depends on stiff (*)
   1: fact: senseclust (0.0.1) depends on paramiko (^2.4)
   1: fact: senseclust (0.0.1) depends on expcomb (*)
   1: fact: senseclust (0.0.1) depends on snakemake (^5.4)
   1: selecting senseclust (0.0.1 bbd609e)
   1: fact: senseclust (0.0.1) depends on nltk (*)
   1: fact: senseclust (0.0.1) depends on sklearn (*)
   1: fact: senseclust (0.0.1) depends on numpy (*)
   1: fact: senseclust (0.0.1) depends on scipy (*)
   1: fact: senseclust (0.0.1) depends on matplotlib (*)
   1: fact: senseclust (0.0.1) depends on click (>=7)
   1: fact: senseclust (0.0.1) depends on finntk (>=0.0.14)
   1: fact: senseclust (0.0.1) depends on fastcluster (*)
   1: fact: senseclust (0.0.1) depends on networkx (*)
   1: fact: senseclust (0.0.1) depends on wikiparse (*)
   1: fact: senseclust (0.0.1) depends on stiff (*)
   1: fact: senseclust (0.0.1) depends on paramiko (^2.4)
   1: fact: senseclust (0.0.1) depends on expcomb (*)
   1: fact: senseclust (0.0.1) depends on snakemake (^5.4)
...

(forever)

Expected output:
An error message (e.g. https://github.com/frankier/finn-sense-clust.git does not provide package finn-sense-clust)

**Comment by stevegore** [comment]
We're having this same issue when we add a local package as a dependency. The loop occurs when a dependency itself has a `pyproject.toml` file. Very frustrating.

**Comment by sjdines** [comment]
Was there a known work around for this?

**Comment by frankier** [comment]
Find and correct the module name in `pyproject.toml`.

**Comment by stevegore** [comment]
Looking back at my comment, I'm not sure my problem was related. We had a submodule that contained a `pyproject.toml` file that was going Poetry to get stuck in an infinite loop. We fixed it by not using submodules.

**Comment by frankier** [comment]
Should be a separate bug then I think.

**Comment by abn** [comment]
@frankier with the fix in #1327 attempting to add the project will produce the following output.

```
Updating dependencies
Resolving dependencies... (2.0s)
                                                                                                 
[RuntimeError]                                                                    
The dependency name for finn-sense-clust does not match the actual package's name: senseclust  
                                                                                                 
add [-D|--dev] [--git GIT] [--path PATH] [-E|--extras EXTRAS] [--optional] [--python PYTHON] [--platform PLATFORM] [--allow-prereleases] [--dry-run] [--] <name> (<name>)...
```
